sycl/include/CL/sycl/queue.hpp

```diff
@@ -214,6 +218,94 @@ class __SYCL_EXPORT queue {
   template <info::queue param>
   typename info::param_traits<info::queue, param>::return_type get_info() const;
 
+private:
+// FIXME remove __NVPTX__ condition once devicelib supports CUDA
+#if !defined(SYCL_DISABLE_FALLBACK_ASSERT) && !defined(__NVPTX__)
+#define __SYCL_ASSERT_START 1
+
+  /**
+   * Submit copy task for assert failure flag and host-task to check the flag
+   * \param Event kernel's event to depend on i.e. the event represents the
+   *              kernel to check for assertion failure
+   * \param SecondaryQueue secondary queue for submit process, null if not used
+   * \returns host tasks event
+   */
+  event submitAssertCapture(event &Event, queue *SecondaryQueue,
+                            const detail::code_location &CodeLoc) {
+    _CODELOCARG(&CodeLoc);
+
+    using AHBufT = buffer<detail::AssertHappened, 1>;
+
+    AHBufT *Buffer = new AHBufT{range<1>{1}};
```

Comment:
Can we implement some recycling mechanism so the same buffer is reused for multiple kernel invocations?

Reply:
I believe, this is feasible. Though, I think this should go in a separate patch.

Reply:
Actually, reuse of the buffer eliminates the deadlock when interfered with host task and the thread pool is of size 1. So, one can see reusing in a02420d